### PR TITLE
[WIP] PATE Example

### PR DESCRIPTION
**Pull request by kamathhrishi**

### CODE REVIEW

File: examples/tutorials/advanced/PATE/Main.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Required imports
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/Main.py]: Main.py -> main.py

File: examples/tutorials/advanced/PATE/Main.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+print("Training Student")
+
+print("\n")
+print("\n")
+
+# Split the test data further into training and validation data for student
+train, val = split(test_loader, args.batchsize)
+
+student = Student(args, Model())
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/Main.py]: add comments about what's going on here

File: examples/tutorials/advanced/PATE/Main.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+student.train(N)
+
+results = []
+targets = []
+
+total = 0.0
+correct = 0.0
+
+for data, target in val:
+
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/Main.py]: add comments

File: examples/tutorials/advanced/PATE/Main.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    predict_lol = student.predict(data)
+    correct += float((predict_lol == (target)).sum().item())
+    total += float(target.size(0))
+
+print("Private Baseline: ", (correct / total) * 100)
+
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/Main.py]: add comments

File: examples/tutorials/advanced/PATE/Main.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+counts_lol = torch.stack(counts).contiguous().view(50, 10000)
+predict_lol = torch.tensor(predict).view(10000)
+
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/Main.py]: add comments

File: examples/tutorials/advanced/PATE/README.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# PATE Example
+
+The scripts in the folder allow you to train a MNIST model using PATE diffrential privacy framework.
+
+```
+$ python Main.py
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/README.md]: Main.py -> main.py

File: examples/tutorials/advanced/PATE/util.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+
+def plot(x, y):
+    """Plots a graph of given x and y.
+       Args:
+           
+           x:
+           y:
+    """
+    pass
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/util.py]: ?

File: examples/tutorials/advanced/PATE/util.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+
+def histogram(x, y):
+    """Plots a histogram for corresponding x and y:
+       Args:
+           
+           x:
+           y:
+    """
```
Comment (mari-linhares) [inline examples/tutorials/advanced/PATE/util.py]: ?

Comment (kamathhrishi) [reply examples/tutorials/advanced/PATE/util.py]: This was to plot the histogram which I probably won't need but in the future.

File: examples/tutorials/advanced/PATE/util.py (hunk 7 above)
Comment (kamathhrishi) [reply examples/tutorials/advanced/PATE/util.py]: Same as histogram

**Comment by mari-linhares** [comment]
@kamathhrishi I'll try to re-review it later today!

**Comment by kamathhrishi** [comment]
@mari-linhares barely made any changes. I will let you know when you should re-review it.

**Comment by robert-wagner** [comment]
@kamathhrishi What is happening with this pr?

**Comment by kamathhrishi** [comment]
@robert-wagner , still in progress :/

**Comment by iamtrask** [comment]
Given low activity and @kamathhrishi 's last post - I'm changing the title to [WIP] until the functionality is correct (just so signal that there's no need to review the PR at this time)

Best of luck!

**Pull request closed without merging**